### PR TITLE
zynqmp: platform: provide uart configuration during compilation

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add possibility to provide UART configuration as a compile
flag (CFG_UART_BASE, CFG_UART_IT, CFG_UART_CLK_HZ).

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/plat-zynqmp/conf.mk [hunk 1]
```diff
@@ -34,6 +39,11 @@ endif
 endif
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),ultra96))
+
+CFG_UART_BASE ?= UART1_BASE
+CFG_UART_IT ?= IT_UART1
+CFG_UART_CLK_HZ ?= UART1_CLK_IN_HZ
```
[the remark this answers is not in the document]
Comment (igoropaniuk) [reply core/arch/arm/plat-zynqmp/conf.mk:45]: @etienne-lms I've used the same config option name, as currently used for other platforms:
```
$ grep -e UART_BASE -r ./
./core/arch/arm/plat-mediatek/main.c:			CONSOLE_UART_BASE, SERIAL8250_UART_REG_SIZE);
./core/arch/arm/plat-mediatek/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-mediatek/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-mediatek/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-mediatek/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-mediatek/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-marvell/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-marvell/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-marvell/main.c:	mvebu_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-marvell/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
./core/arch/arm/plat-marvell/platform_config.h:#define PLAT_MARVELL_BOOT_UART_BASE		(MVEBU_REGS_BASE + 0x512000)
./core/arch/arm/plat-marvell/platform_config.h:#define CONSOLE_UART_BASE	PLAT_MARVELL_BOOT_UART_BASE
./core/arch/arm/plat-marvell/platform_config.h:#define PLAT_MARVELL_BOOT_UART_BASE		(MVEBU_REGS_BASE + 0x12000)
./core/arch/arm/plat-marvell/platform_config.h:#define CONSOLE_UART_BASE	PLAT_MARVELL_BOOT_UART_BASE
./core/arch/arm/plat-marvell/platform_config.h:#define PLAT_MARVELL_BOOT_UART_BASE		0x87E028000000ll
./core/arch/arm/plat-marvell/platform_config.h:#define CONSOLE_UART_BASE			PLAT_MARVELL_BOOT_UART_BASE
./core/arch/arm/plat-synquacer/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-synquacer/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
./core/arch/arm/plat-synquacer/platform_config.h:#define CONSOLE_UART_BASE		UART0_BASE
./core/arch/arm/plat-rcar/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, SCIF_REG_SIZE);
./core/arch/arm/plat-rcar/main.c:	scif_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-rcar/platform_config.h:#define CONSOLE_UART_BASE	0xE6E88000
./core/arch/arm/plat-rcar/platform_config.h:#define CONSOLE_UART_BASE	0xE6C50000
./core/arch/arm/plat-rcar/platform_config.h:#define CONSOLE_UART_BASE	0xE6540000
./core/arch/arm/plat-aspeed/platform_ast2600.c:register_phys_mem(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE, SMALL_PAGE_SIZE);
./core/arch/arm/plat-aspeed/platform_ast2600.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-aspeed/platform_config.h:#define CONSOLE_UART_BASE	UART5_BASE
./core/arch/arm/plat-stm32mp1/platform_config.h:#define CONSOLE_UART_BASE		STM32MP1_DEBUG_USART_BASE
./core/arch/arm/plat-rzn1/main.c:	ns16550_init(&console_data, CONSOLE_UART_BASE, IO_WIDTH_U32, 2);
./core/arch/arm/plat-rzn1/platform_config.h:#define CONSOLE_UART_BASE		0x40060000
./core/arch/arm/plat-sunxi/main.c:#ifdef CONSOLE_UART_BASE
./core/arch/arm/plat-sunxi/main.c:			CONSOLE_UART_BASE, SUNXI_UART_REG_SIZE);
./core/arch/arm/plat-sunxi/main.c:			     CONSOLE_UART_BASE,
./core/arch/arm/plat-sunxi/platform_config.h:#define CONSOLE_UART_BASE	0x01c28000 /* UART0 */
./core/arch/arm/plat-poplar/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE, PL011_REG_SIZE);
./core/arch/arm/plat-poplar/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-poplar/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART0_BASE
./core/arch/arm/plat-bcm/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-bcm/platform_config.h:#define CONSOLE_UART_BASE	0x68a10000
./core/arch/arm/plat-totalcompute/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, PL011_REG_SIZE);
./core/arch/arm/plat-totalcompute/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
./core/arch/arm/plat-totalcompute/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-ti/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-ti/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-ti/platform_config.h:#define CONSOLE_UART_BASE       UART1_BASE
./core/arch/arm/plat-ti/platform_config.h:#define CONSOLE_UART_BASE       UART3_BASE
./core/arch/arm/plat-ti/platform_config.h:#define CONSOLE_UART_BASE       UART0_BASE
./core/arch/arm/plat-k3/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-k3/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-k3/platform_config.h:#define CONSOLE_UART_BASE       (UART0_BASE + CFG_CONSOLE_UART * 0x10000)
./core/arch/arm/plat-hisilicon/main.c:register_phys_mem(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE, PL011_REG_SIZE);
./core/arch/arm/plat-hisilicon/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-hisilicon/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART0_BASE
./core/arch/arm/plat-sprd/console.c:	sprd_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-sprd/main.c:			ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_PGDIR_SIZE),
./core/arch/arm/plat-sprd/platform_config.h:#define CONSOLE_UART_BASE	UART1_BASE
./core/arch/arm/plat-versal/main.c:			ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_PGDIR_SIZE),
./core/arch/arm/plat-versal/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-versal/platform_config.h:#define CONSOLE_UART_BASE	UART0_BASE
./core/arch/arm/plat-rzg/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, SCIF_REG_SIZE);
./core/arch/arm/plat-rzg/main.c:	scif_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-rzg/platform_config.h:#define CONSOLE_UART_BASE	0xE6E88000
./core/arch/arm/plat-ls/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-ls/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, 0, 0);
./core/arch/arm/plat-ls/main.c:	ns16550_init(&console_data, CONSOLE_UART_BASE, IO_WIDTH_U8, 0);
./core/arch/arm/plat-ls/platform_config.h:#define CONSOLE_UART_BASE		UART0_BASE
./core/arch/arm/plat-rpi3/main.c:			CONSOLE_UART_BASE, SERIAL8250_UART_REG_SIZE);
./core/arch/arm/plat-rpi3/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-rpi3/platform_config.h:#define CONSOLE_UART_BASE	0x3f215040 /* UART0 */
./core/arch/arm/plat-uniphier/main.c:			ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_PGDIR_SIZE),
./core/arch/arm/plat-uniphier/main.c:	serial8250_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-uniphier/platform_config.h:#define UART_BASE		0x54006800
./core/arch/arm/plat-uniphier/platform_config.h:#define CONSOLE_UART_BASE	(UART_BASE + 0x100 * UART_CH)
./core/arch/arm/plat-sam/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-sam/main.c:	atmel_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-sam/platform_config.h:#define CONSOLE_UART_BASE     AT91C_BASE_UART1
./core/arch/arm/plat-zynqmp/conf.mk:CFG_UART_BASE ?= UART0_BASE
./core/arch/arm/plat-zynqmp/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-zynqmp/main.c:			ROUNDDOWN(CONSOLE_UART_BASE, CORE_MMU_PGDIR_SIZE),
./core/arch/arm/plat-zynqmp/main.c:	cdns_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-zynqmp/platform_config.h:#define CONSOLE_UART_BASE		(CFG_UART_BASE)
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART5_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART4_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART4_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART4_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART2_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART5_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART7_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART2_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART3_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART2_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART2_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART3_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART0_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART0_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART5_BASE
./core/arch/arm/plat-imx/conf.mk:CFG_UART_BASE ?= UART1_BASE
./core/arch/arm/plat-imx/main.c:#ifdef CONSOLE_UART_BASE
./core/arch/arm/plat-imx/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-imx/main.c:#ifdef CONSOLE_UART_BASE
./core/arch/arm/plat-imx/main.c:	imx_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-imx/platform_config.h:#define CONSOLE_UART_BASE		(CFG_UART_BASE)
./core/arch/arm/plat-amlogic/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-amlogic/main.c:	amlogic_uart_init(&console_data, CONSOLE_UART_BASE);
./core/arch/arm/plat-amlogic/platform_config.h:#define CONSOLE_UART_BASE	0xFF803000
./core/arch/arm/plat-vexpress/main.c:register_phys_mem_pgdir(MEM_AREA_IO_SEC, CONSOLE_UART_BASE, PL011_REG_SIZE);
./core/arch/arm/plat-vexpress/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
./core/arch/arm/plat-vexpress/platform_config.h:#define CONSOLE_UART_BASE	UART1_BASE
./core/arch/arm/plat-vexpress/platform_config.h:#define CONSOLE_UART_BASE	UART3_BASE
./core/arch/arm/plat-vexpress/platform_config.h:#define CONSOLE_UART_BASE	UART1_BASE
./core/arch/arm/plat-vexpress/platform_config.h:#define CONSOLE_UART_BASE	UART1_BASE
./core/arch/arm/plat-d02/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-d02/main.c:	hi16xx_uart_init(&console_data, CONSOLE_UART_BASE,
./core/arch/arm/plat-d02/platform_config.h:#define CONSOLE_UART_BASE       (PERI_SUB_CTRL_ADDR + 0x00300000)
./core/arch/arm/plat-hikey/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE, PL011_REG_SIZE);
./core/arch/arm/plat-hikey/main.c:	pl011_init(&console_data, CONSOLE_UART_BASE, CONSOLE_UART_CLK_IN_HZ,
./core/arch/arm/plat-hikey/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART3_BASE
./core/arch/arm/plat-hikey/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART2_BASE
./core/arch/arm/plat-hikey/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART0_BASE
./core/arch/arm/plat-hikey/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART6_BASE
./core/arch/arm/plat-hikey/platform_config.h:#define CONSOLE_UART_BASE	PL011_UART5_BASE
./core/arch/arm/plat-zynq7k/main.c:register_phys_mem_pgdir(MEM_AREA_IO_NSEC, CONSOLE_UART_BASE,
./core/arch/arm/plat-zynq7k/main.c:	cdns_uart_init(&console_data, CONSOLE_UART_BASE, 0, 0);
./core/arch/arm/plat-zynq7k/platform_config.h:#define CONSOLE_UART_BASE		UART1_BASE

```

If we change it here, that also makes sense to do that for all other platforms.
As I mentioned in the previous comment, I do agree with all these changes, I'd prefer to do that in a separate PR.

**Comment by igoropaniuk** [comment]
Applied @jenswi-linaro @jforissier A-b tags

@etienne-lms let me know if you're ok about my suggestions in the previous comments

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` even if I still think config switch names should be changed.

**Pull request merged**